### PR TITLE
Conda forge package kmc_smudge_pairs

### PR DESCRIPTION
**Pull request by gkanogiannis**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by gkanogiannis** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by gkanogiannis** [comment]
@conda-forge/staged-recipes, ready for review.

**Comment by gkanogiannis** [comment]
Gently asking for a review @conda-forge/core @conda-forge/help-c-cpp

**Comment by gkanogiannis** [comment]
This is 3 weeks here...
@conda-forge-admin, please ping conda-forge/help-c-cpp

**Comment by gkanogiannis** [comment]
@conda-forge-admin, please ping team.

**Comment by gkanogiannis** [comment]
@conda-forge-admin, please ping conda-forge/help-c-cpp

File: recipes/kmc-smudge-pairs/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/gkanogiannis/KMC/archive/v{{ version }}.tar.gz
+  sha256: d6acc23643e3764a5ce6e3ec4cd1790592f36d992e44dc927baf0cbf14842f46
+
+build:
+  skip: true  # [not linux]
+  script: bash build_static.sh
```
Comment (xhochy) [inline recipes/kmc-smudge-pairs/meta.yaml:14]: We would like to have shared builds on conda-forge.

File: recipes/kmc-smudge-pairs/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - make
+  host:
+    - zlib
+    - bzip2
+  run:
+    - zlib
+    - bzip2
```
Comment (xhochy) [inline recipes/kmc-smudge-pairs/meta.yaml:27]: This should be handled by run_exports.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/kmc-smudge-pairs/meta.yaml (hunk 1 above)
Comment (gkanogiannis) [reply recipes/kmc-smudge-pairs/meta.yaml]: Hi @xhochy, thank you for reviewing this. I changed the code and makefile to build by linking to shared libraries (instead static .a provided for zlib and bzip2). The problem now is with the docker build that lacks these libraries. I get error "fatal error: zlib.h: No such file or directory". Any advice how to overcome this? Thank you for your support.

File: recipes/kmc-smudge-pairs/meta.yaml (hunk 2 above)
Comment (gkanogiannis) [reply recipes/kmc-smudge-pairs/meta.yaml]: I moved these libraries to the run_exports section of build.

File: recipes/kmc-smudge-pairs/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - make
+    - libzlib
+    - anaconda::zlib-devel-amzn2-aarch64
+    - bzip2
+  host:
+    - zlib
+    - bzip2
```
Comment (ForrestQuant) [inline recipes/kmc-smudge-pairs/meta.yaml:30]: Don't have them at all here. Once, this is removed, I'm happy to review the error message and give guidance on how to continue.
```suggestion
  host:
    - zlib
    - bzip2
```

File: recipes/kmc-smudge-pairs/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - make
+    - libzlib
+    - anaconda::zlib-devel-amzn2-aarch64
+    - bzip2
```
Comment (xhochy) [inline recipes/kmc-smudge-pairs/meta.yaml:27]: Please remove these, once this has been updated, I'm happy to review the next error message and give guidance.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/kmc-smudge-pairs/meta.yaml (hunk 4 above)
Comment (gkanogiannis) [reply recipes/kmc-smudge-pairs/meta.yaml]: Removed. Thank you for your guidance.

**Comment by xhochy** [comment]
The problem is that `$CXXFLAGS` is not passed on to the compiler command. You will need to respect that (and probably `$LDFLAGS`)  to make the builds pass.

**Comment by gkanogiannis** [comment]
> The problem is that `$CXXFLAGS` is not passed on to the compiler command. You will need to respect that (and probably `$LDFLAGS`) to make the builds pass.

I am not sure what you mean. I can locally build it, with the same makefile and all. But at my local machine I have installed bzip2 and zlib headers, which seem to be missing from the docker builders here.

**Comment by gkanogiannis** [comment]
Oh, got it! Apologies, it is my first try with conda-forge. Could you please direct me to some resource on what info to pass to the compiler and linker? Or maybe a similar package that I could take a look? Thank you again.

**Comment by xhochy** [comment]
The above flags should be sufficent. Most other packages use a build system like CMake or Meson where these flags are picked up automatically.

**Comment by gkanogiannis** [comment]
Dear @xhochy, I think I managed to get it through and build succesfully.

File: recipes/kmc-smudge-pairs/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+source:
+  url: https://github.com/gkanogiannis/KMC/archive/v{{ version }}.tar.gz
+  sha256: 43ae55aabc1334a038e5898b2fdc9a1d254d6a434c9ad34e56663be1aec9dc42
+
+build:
+  skip: true  # [not linux]
+  number: 0
+  run_exports:
+    - zlib
+    - bzip2
```
Comment (xhochy) [inline recipes/kmc-smudge-pairs/meta.yaml:17]: This is in the packages themselves and should not be here.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/kmc-smudge-pairs/meta.yaml (hunk 5 above)
Comment (gkanogiannis) [reply recipes/kmc-smudge-pairs/meta.yaml]: This was removed.

**Review by xhochy — APPROVED** [review]

**Pull request merged**